shell/plugins/steve/caches/base-cache.js

```diff
@@ -0,0 +1,296 @@
+    this.calculatedFields.forEach(({ name, func }) => {
+      const calculatedField = { [name]: func({ ...resource, ...calculatedFields }, this.getters, this.rootGetters) };
+
+      Object.assign(calculatedFields, calculatedField); // We'll keep calculatedFields when this function finishes
+    });
+
+    // passing them back in two pieces means we keep the resource in it's original condition but still have the calculatedFields for searching/sorting operations
+    return calculatedFields;
+  }
+
+  __updateCache() {
+    console.warn(`Cache Class ${ this.type } did not specify a method for updating cache, nothing happened`); // eslint-disable-line no-console
+  }
+
+  /**
+   * Requests data from the cache's data source
+   */
+  async request() {
+    await setTimeout(() => console.warn(`Cache Class ${ this.type } did not specify a method for requesting cache content from its source, nothing happened`), 0); // eslint-disable-line no-console
+
+    return this;
+  }
+
+  /**
+   * Sets the current cache with the payload
+   */
+  load() {
+    console.warn(`Cache Class ${ this.type } did not specify a method for loading cache, nothing happened`); // eslint-disable-line no-console
+
+    return this;
+  }
+
+  // checks the cache's currentParams against the new params and determines if a new request would be required to re-request from the API in order to satisfy it
+  cacheIsInvalid() {
+    console.warn(`Cache Class ${ this.type } did not specify a method for checking cache validity, nothing happened`); // eslint-disable-line no-console
+
+    return true;
+  }
+
+  // get the full list of resources with both the raw resource and the whole resource (with calculated fields)
+  resourceList() {
+    return Object.values(this.resources)
+      .map(({ resource, calculatedFields }) => {
+        const resourceCombo = {
+          resource,
+          wholeResource: {
+            ...resource,
+            ...calculatedFields
+          }
+        };
+
+        return resourceCombo;
+      });
+  }
+
+  // gets the full list of wholeResources as an array from the cache
+  all(returnWholeResource = true) {
+    return this.resourceList()
+      .map(({ resource, wholeResource }) => {
+        return returnWholeResource ? wholeResource : resource;
+      });
+  }
+
+  // gets a specific id from the cache and returns it as a whole resource if it exists
+  byId(id, wholeResource = true) {
+    const resource = this.resources[id];
+
+    if (resource) {
+      return {
+        ...resource?.resource,
+        ...(wholeResource ? calculatedFields : {}),
+      };
+    }
+  }
+
+  // gets a specific list of ids from the cache and returns it as a whole resource if it exists
+  byIds(ids, wholeResource = true) {
+    return ids
+      .map(id => this.byId(id, wholeResource))
+      .filter(resource => resource);
+  }
+
+  /**
+   * Find the resource/s associated with the params in the cache. IF we don't have the resource/s for the params we'll fetch them
```

Comment:
Given this method now doesn't make the http request
- jsdoc needs updating
- `find` name might be better renamed to `get` (to avoid confusion with the find = http request + get from cache concept)